app/lib/backup/runner.ts

```diff
@@ -215,12 +229,25 @@ const encodeAndEncrypt = <T>(ctx: Context, data: T) =>
 
 const toMessageData = (message: Api.Message | Api.MessageService): MessageData | ServiceMessageData => {
   let from
-  if (message.fromId?.className === 'PeerUser') {
-    from = message.fromId.userId.toString()
-  } else if (message.fromId?.className === 'PeerChat') {
-    from = message.fromId.chatId.toString()
-  } else if (message.fromId?.className === 'PeerChannel') {
-    from = message.fromId.channelId.toString()
+  let fromEntity
+
+  if(message.fromId){ 
+    fromEntity = message.fromId
+  } else {
+    /**
+     * Note: 
+     *   In the context of a private chat with another user, the `fromId` can be undefined. 
+     *   In such cases, the `peerId` represents the user our Telegram account is communicating with.
+     */
+    fromEntity = message.peerId
```

Comment:
Just add `peer` as a field on `MessageData` in addition to `from` and collect this. We shouldn't change the telegram schema, just work with that we have.